vendor/bat-native-ledger/src/bat/ledger/internal/bat_helper.cc

```diff
@@ -1104,13 +1104,16 @@ bool RECONCILE_DIRECTION::loadFromJson(const std::string & json) {
   // has parser errors or wrong types
   bool error = d.HasParseError();
   if (!error) {
-    error = !(d.HasMember("amount") && d["amount"].IsInt() &&
+    // An older serialization format stored the amount as an int, so to retain
+    // backward compatibility, allow int or double.
+    error = !(d.HasMember("amount") &&
+      (d["amount"].IsInt() || d["amount"].IsDouble()) &&
       d.HasMember("publisher_key") && d["publisher_key"].IsString() &&
       d.HasMember("currency") && d["currency"].IsString());
   }
 
   if (!error) {
-    amount_ = d["amount"].GetInt();
+    amount_ = d["amount"].GetDouble();
```

Comment:
would this work with an old profile? Or do we need to detect if int and get it as int and then cast it to double?